main/lsp/requests/hover.cc

```diff
@@ -14,22 +14,18 @@ namespace sorbet::realmain::lsp {
 string methodInfoString(const core::GlobalState &gs, const core::TypePtr &retType,
                         const core::DispatchResult &dispatchResult,
                         const unique_ptr<core::TypeConstraint> &constraint) {
-    string contents;
+    set<string> contents;
     auto start = &dispatchResult;
     ;
     while (start != nullptr) {
         auto &component = start->main;
         if (component.method.exists()) {
-            if (!contents.empty()) {
-                contents += "\n";
-            }
-            contents = absl::StrCat(
-                contents, prettyTypeForMethod(gs, component.method, component.receiver, retType, constraint.get()));
+            contents.insert(prettyTypeForMethod(gs, component.method, component.receiver, retType, constraint.get()));
         }
         start = start->secondary.get();
     }
 
-    return contents;
+    return absl::StrJoin(contents, "\n");
```

Comment:
This will return the type descriptions in sorted order of the type descriptions, which might not correspond to the way the user wrote the type...though it's hard once you introduce deduplication into the picture to get "close" to what the user was writing.

WDYT about changing the code (along with testcases) so that it returns the type descriptions in the (deduplicated) order that they appeared in the `dispatchResult`?

I'm not sure offhand it it would make a difference.

Reply:
Ahh, good call out! That's my misunderstanding. It's been awhile since I used `std::set`, I had assumed that it maintained the order that the items were inserted into it. 

I had been aiming for the order that you've requested and agree that this makes the most sense. I shall update the code to match that now.

Reply:
> the way the user wrote the type

hm, I'm not sure it makes sense to consider anything about "the order the user wrote the types" here

the thing that's showing up in the `resultType` of the dispatch result is a computed value (which might be the result of unioning or intersecting arbitrary types, arriving at an entirely new type). In this example yes because it's a simple union of class types there's a clear order, but in general types might collapse or combine in ~arbitrary ways.